File: src/cpp/generate.cpp
#include <iostream>
#include <filesystem>

using namespace std;

#define int long long

string to_4digit(int n) {
    string s = to_string(n);
    while (s.size() < 4) {
        s = "0" + s;
    }
    return s;
}

signed main() {

    string s;
    int k;
    #ifdef _WIN32
        system("g++ -o application input.cpp");
    #else
        system("g++ -o application.o input.cpp");
    #endif


    int starting_test, ending_test;
    cout << "Starting test: "; cin >> starting_test;
    if(starting_test < 1){
        return cout << "Error of number starting test" << endl, 0;
    }
    cout << "Ending test: "; cin >> ending_test;
    if(ending_test > 200){
        return cout << "Error of number ending test" << endl, 0;
    }


    string input_file;
    for(int i = starting_test; i <= ending_test; ++i){
        input_file = "tests/" + to_4digit(i) + ".in";

        #ifdef _WIN32
            s = ".\\application.exe > " + input_file;
        #else 
            s = "./application.o > " + input_file;
        #endif

        const char *arr = s.c_str();
        int k = system(arr);

        if(k == 0){
            cout << "Test " << to_4digit(i) << " OK" << endl;
        } else {
            cout << "Test " << to_4digit(i) << " ERROR Code " << k << endl;
        }
    }
}

File: src/python/generate.py
import os

# check if the folder is not exists
if not os.path.exists("tests"):
    os.makedirs("tests")

s = ""
k = 0
starting_test = int(input("Starting test: "))
if starting_test < 1:
    print("Error of number starting test")
    exit(0)

ending_test = int(input("Ending test: "))
if ending_test < 1:
    print("Error of number ending test")
    exit(0)

input_file = ""
for i in range(starting_test, ending_test+1):
    input_file = f"tests/{i:04}.in"
    if os.name == 'nt':
        s = f"py input.py > {input_file}"
    else:
        s = f"python3 input.py > {input_file}"
    
    k = os.system(s)

    if k == 0:
        print(f"Test {i:04} OK")
    else:
        print(f"Test {i:04} ERROR Code {k}")
